src/main/java/com/app/store/rest/BookController.java

```diff
@@ -53,8 +59,11 @@ public BookController(BookService bookService) {
    * @return the book
    */
   @PostMapping
-  public Book create(@RequestBody Book newbook) {
-    return bookService.create(newbook);
+  public ResponseEntity<Book> create(@RequestBody Book newbook) {
+    Book book =bookService.create(newbook);
+    URI location = ServletUriComponentsBuilder.fromCurrentRequest().path("").buildAndExpand().toUri();
+      return ResponseEntity.created(location).body(book);
```

Comment:
https://docs.spring.io/spring-hateoas/docs/1.2.4/reference/html/#server.representation-model-assembler,
keep you code as it is but maybe in golive project we will use representation-model-assembler this is the doc if you want to take a look